pkg/security/securitycontextconstraints/provider.go

```diff
@@ -292,19 +292,38 @@ func (s *simpleProvider) ValidateContainerSecurityContext(pod *api.Pod, containe
 
 	allErrs = append(allErrs, s.capabilitiesStrategy.Validate(pod, container)...)
 
-	if len(pod.Spec.Volumes) > 0 && !sccutil.SCCAllowsAllVolumes(s.scc) {
-		allowedVolumes := sccutil.FSTypeToStringSet(s.scc.Volumes)
-		for i, v := range pod.Spec.Volumes {
-			fsType, err := sccutil.GetVolumeFSType(v)
-			if err != nil {
-				allErrs = append(allErrs, field.Invalid(fldPath.Child("volumes").Index(i), string(fsType), err.Error()))
-				continue
+	if len(pod.Spec.Volumes) > 0 {
+		if !sccutil.SCCAllowsAllVolumes(s.scc) {
+			allowedVolumes := sccutil.FSTypeToStringSet(s.scc.Volumes)
+			for i, v := range pod.Spec.Volumes {
+				fsType, err := sccutil.GetVolumeFSType(v)
+				if err != nil {
+					allErrs = append(allErrs, field.Invalid(fldPath.Child("volumes").Index(i), string(fsType), err.Error()))
+					continue
+				}
+
+				if !allowedVolumes.Has(string(fsType)) {
+					allErrs = append(allErrs, field.Invalid(
+						fldPath.Child("volumes").Index(i), string(fsType),
+						fmt.Sprintf("%s volumes are not allowed to be used", string(fsType))))
+				}
 			}
+		}
 
-			if !allowedVolumes.Has(string(fsType)) {
-				allErrs = append(allErrs, field.Invalid(
-					fldPath.Child("volumes").Index(i), string(fsType),
-					fmt.Sprintf("%s volumes are not allowed to be used", string(fsType))))
+		if len(s.scc.AllowedFlexDrivers) > 0 && sccutil.SCCAllowsFSType(s.scc, securityapi.FSTypeFlexVolume) {
+			for i, v := range pod.Spec.Volumes {
+				if v.FlexVolume == nil {
+					continue
+				}
+
+				for _, allowedDriver := range s.scc.AllowedFlexDrivers {
+					driver := v.FlexVolume.Driver
+					if driver != allowedDriver {
```

Comment:
doesn't this mean that if I allow `["foo", "bar"]` and request one with `"bar"`, I'll get an error even though I'm allowed?  How about adding some tests here.